lib/apis/v3/ippool.go

```diff
@@ -45,7 +45,7 @@ type IPPoolSpec struct {
 
 	// Contains configuration for VXLAN tunneling for this pool. If not specified,
 	// then this is defaulted to "Never" (i.e. VXLAN tunelling is disabled).
-	VXLANMode VXLANMode `json:"vxlanMode,omitempty" validate:"omitempty,vxlanMode"`
+	VXLANMode VXLANMode `json:"vxlanMode,omitempty" validate:"omitempty,vxlanMode" default:"Never"`
```

Comment:
@asincu What is interpreting this tag? I don't see any other fields that have this on them

Reply:
There's a bit of indirection here; it will get picked up by `go-defaults` ([context](https://github.com/projectcalico/calicoctl/pull/2006/files)).

As per your comment:

> I'm a little bit hesitant to introduce a new library here [...]

it certainly doesn't feel worthwhile if we do this only for this one property.

Reply:
@jpalardy @caseydavenport I would propose that we extend the usage of this tag for all fields that require a different default than the ones that go provides.

Reply:
If there are other use cases, converting them now (in a follow up commit) is an option, but the cool thing is that you do not need to, you can start using it gradually.

There are certainly other cases that might benefit from a reasonable default, e.g. policy order to be set to 0 instead of nil (i just noticed last week). That would also make the calicoctl output nicer, however, is is easily doable using this library?

Reply:
@caseydavenport @jpalardy @tomastigera I have added default never to IpIpMode in order for the additional test to pass. Please see https://github.com/projectcalico/calicoctl/pull/2006/commits/84103203b48bb07011cdf44ffe2638b5408eca46